pkg/loader/load.go

```diff
@@ -444,6 +450,79 @@ func (s *loaderImpl) execDDL(ddl *DDL) error {
 	return errors.Trace(err)
 }
 
+func (s *loaderImpl) processOracleDDL(ddl *DDL) error {
+	ddlStmt := ddl.SQL
+	newStmt := ""
+	if isTruncateTableStmt(ddlStmt) {
+		newStmt = fmt.Sprintf("BEGIN %s.do_truncate('%s.%s','');END;", ddl.Database, ddl.Database, ddl.Table)
+	} else {
+		ok, astStmt := isTruncateTablePartitionStmt(ddlStmt)
+		if ok {
+			partitions := astStmt.Specs[0].PartitionNames
+			newStmt = fmt.Sprintf("BEGIN %s.do_truncate('%s.%s','%s');END;", ddl.Database, ddl.Database, ddl.Table, partitions[0].O)
+		} else {
+			log.Warn(fmt.Sprintf("oracle do not support ddl[%s]", ddlStmt))
+			return nil
+		}
+	}
+
+	err := util.RetryContext(s.ctx, maxDDLRetryCount, execDDLRetryWait, 1, func(context.Context) error {
+		newStmt := newStmt
+		tx, err := s.db.Begin()
+		if err != nil {
+			return err
+		}
+		if _, err = tx.Exec(newStmt); err != nil {
+			log.Error("DDL exec failed.", zap.String("old sql", ddl.SQL), zap.String("new ddl sql", newStmt), zap.Error(err))
+			if rbErr := tx.Rollback(); rbErr != nil {
+				log.Error("Rollback DDL failed", zap.String("old sql", ddl.SQL), zap.String("new ddl sql", newStmt), zap.Error(rbErr))
+			}
+			return err
+		}
+
+		if err = tx.Commit(); err != nil {
+			return err
+		}
```

Comment:
Don't need a txn here. Besides, why `newStmt := newStmt`?
```suggestion
		tx, err := s.db.Exec(newStmt)
```

Reply:
for txn , it is same as mysql ddl process.